src/main/java/org/opentripplanner/routing/core/RoutingRequest.java

```diff
@@ -1131,28 +1153,45 @@ public void setWaitAtBeginningFactor(double waitAtBeginningFactor) {
     public void banTrip(AgencyAndId trip) {
         bannedTrips.put(trip, BannedStopSet.ALL);
     }
-    
-    /** 
-     * tripIsBanned is a misnomer: this checks whether the agency or route are banned.
-     * banning of individual trips is actually performed inside the trip search, 
-     * in TripTimes.tripAcceptable.
-     */
-    public boolean tripIsBanned(Trip trip) {
+
+    public boolean routeIsBanned(Route route) {
         /* check if agency is banned for this plan */
         if (bannedAgencies != null) {
-            if (bannedAgencies.contains(trip.getRoute().getAgency().getId())) {
+            if (bannedAgencies.contains(route.getAgency().getId())) {
                 return true;
             }
         }
 
         /* check if route banned for this plan */
         if (bannedRoutes != null) {
-            Route route = trip.getRoute();
             if (bannedRoutes.matches(route)) {
                 return true;
             }
         }
 
+        boolean whiteListed = false;
+        boolean whiteListInUse = false;
+
+        /* check if agency is whitelisted for this plan */
+        if (whiteListedAgencies != null && whiteListedAgencies.size() > 0) {
+            whiteListInUse = true;
+            if (whiteListedAgencies.contains(route.getAgency().getId())) {
+                whiteListed = true;
+            }
+        }
+
+        /* check if route is whitelisted for this plan */
+        if (whiteListedRoutes != null && !whiteListedRoutes.isEmpty()) {
+            whiteListInUse = true;
+            if (whiteListedRoutes.matches(route)) {
+                whiteListed = true;
+            }
+        }
+
+        if (whiteListInUse && !whiteListed) {
```

Comment:
At first glance `whiteListInUse` seems like an unnecessary variable, but it certainly adds to the readability of the boolean expression.